.ebextensions/cloudwatch.config

```diff
@@ -1,26 +1,47 @@
-packages:
-  yum:
-    perl-DateTime: []
-    perl-Sys-Syslog: []
-    perl-LWP-Protocol-https: []
-    perl-Switch: []
-    perl-URI: []
-    perl-Digest-SHA: []
-
-sources:
-  /opt/cloudwatch: https://aws-cloudwatch.s3.amazonaws.com/downloads/CloudWatchMonitoringScripts-1.2.1.zip
-
+files:
+  "/opt/aws/amazon-cloudwatch-agent/bin/config.json":
```

Comment:
I wonder about the config file location, I see things in `/opt/aws/amazon-cloudwatch-agent/etc/amazon-cloudwatch-agent.d` - perhaps better around there? Or perhaps better to keep custom files outside aws managed directories and go in /var/app/current?

Reply:
Yes I wondered about alternate locations. I used the current file location as it was used in this specific example:

https://docs.aws.amazon.com/elasticbeanstalk/latest/dg/customize-containers-cw.html

Running with the current command it validates the configuration, writes to the validation log, then copies it to the amazon-cloudwatch-agent.d directory. So it could potentially start elsewhere, I guess.

The fetch-config and `-s` does a restart of the agent after installing new config, I think if we didn't use fetch-config we'd need to stop/start?